pkg/skaffold/schema/validation/validation.go

```diff
@@ -219,6 +219,72 @@ func validateUniqueDependencyAliases(artifacts []*latest.Artifact) (errs []error
 	return
 }
 
+// extractContainerNameFromNetworkMode returns the container name even if it comes from an Env Var. Error if the mode isn't valid
+// (only container:<id|name> format allowed)
+func extractContainerNameFromNetworkMode(mode string) (string, error) {
+	if strings.HasPrefix(strings.ToLower(mode), "container:") {
+		// Up to this point, we know that we can strip until the colon symbol and keep the second part
+		// this is helpful in case someone sends container not in lowercase
+		maybeID := strings.SplitN(mode, ":", 2)[1]
+		id, err := util.ExpandEnvTemplate(maybeID, map[string]string{})
+		if err != nil {
+			return "", sErrors.NewError(err,
+				proto.ActionableErr{
+					Message: fmt.Sprintf("unable to parse container name %s: %s", mode, err),
+					ErrCode: proto.StatusCode_TEST_CUSTOM_CMD_PARSE_ERR,
```

Comment:
shd we reuse the  `StatusCode_INIT_DOCKER_NETWORK_INVALID_CONTAINER_NAME`  error code and the suggestion code should be in regards to fixing docker container name
I understand you want to define a new error code for the env template parse error. However the error code you are using right now is specific to "TEST" phase.

Either use the  `StatusCode_INIT_DOCKER_NETWORK_INVALID_CONTAINER_NAME`  or define a new one in `INIT` phase e.g. `StatusCode_INIT_DOCKER_NETWORK_PARSE_ERR`
You can reuse the suggestion code.

Reply:
I've already added a couple of new status and suggestion codes